utils/material_plots_2D.py

```diff
@@ -1,80 +1,167 @@
+    parser = argparse.ArgumentParser(description="Material Plotter")
+    parser.add_argument(
+        "--inputFile", "-f", type=str, help="relative path to the input file"
+    )
+    parser.add_argument(
+        "--angleMin",
+        default=6,
+        type=float,
+        help="Minimum eta/theta/cosTheta",
+    )
+    parser.add_argument(
+        "--angleMax",
+        default=6,
+        type=float,
+        help="Maximum eta/theta/cosTheta",
+    )
+    parser.add_argument(
+        "--angleDef",
+        default="eta",
+        choices=["eta", "theta", "cosTheta", "thetaRad"],
+        type=str,
+        help="Angle definition to use: eta, theta, thetaRad or cosTheta; default: eta",
+    )
+    parser.add_argument(
+        "--angleBinning",
+        "-b",
+        default=0.05,
+        type=float,
+        help="Eta/theta/cosTheta bin width",
+    )
+    parser.add_argument(
+        "--nPhiBins",
+        default=100,
+        type=int,
+        help="Number of bins in phi",
+    )
+    parser.add_argument("--x0Max", "-x", default=0.0, type=float, help="Max of x0")
+    parser.add_argument(
+        "--outputDir",
+        "-o",
+        type=str,
+        default="plots",
+        help="Directory to store output files in",
+    )
     args = parser.parse_args()
 
+    output_dir = Path("data") / args.outputDir
+    output_dir.mkdir(
+        parents=True, exist_ok=True
+    )  # Create the directory if it doesn't exist
+
     ROOT.gStyle.SetNumberContours(100)
 
-    f = ROOT.TFile.Open(args.fname, "read")
+    f = ROOT.TFile.Open(fspath(Path(args.inputFile).with_suffix(".root")), "read")
     tree = f.Get("materials")
-    histDict = {}
 
     ROOT.gROOT.SetBatch(1)
 
-    h_x0 = ROOT.TH2F("h_x0","h_x0", int((args.angleMax-args.angleMin)/args.angleBinning),args.angleMin,args.angleMax,args.nPhiBins,-math.pi,math.pi)
-    h_lambda = ROOT.TH2F("h_lambda","h_lambda", int((args.angleMax-args.angleMin)/args.angleBinning),args.angleMin,args.angleMax,args.nPhiBins,-math.pi,math.pi)
-    h_depth = ROOT.TH2F("h_depth","h_depth", int((args.angleMax-args.angleMin)/args.angleBinning),args.angleMin,args.angleMax,args.nPhiBins,-math.pi,math.pi)
+    h_x0 = ROOT.TH2F(
+        "h_x0",
+        "h_x0",
+        int((args.angleMax - args.angleMin) / args.angleBinning),
+        args.angleMin,
+        args.angleMax,
+        args.nPhiBins,
+        -math.pi,
+        math.pi,
+    )
+    h_lambda = ROOT.TH2F(
+        "h_lambda",
+        "h_lambda",
+        int((args.angleMax - args.angleMin) / args.angleBinning),
+        args.angleMin,
+        args.angleMax,
+        args.nPhiBins,
+        -math.pi,
+        math.pi,
+    )
+    h_depth = ROOT.TH2F(
+        "h_depth",
+        "h_depth",
+        int((args.angleMax - args.angleMin) / args.angleBinning),
+        args.angleMin,
+        args.angleMax,
+        args.nPhiBins,
+        -math.pi,
+        math.pi,
+    )
```

Comment:
Apart from the names & titles these are the same histograms, this could be moved into a separate function that only takes the name as argument.

Reply:
good idea. done